### PR TITLE
Bg_Hidan_Fwbig

### PR DESCRIPTION
**Pull request by petrie911**
Even the biggest firewalls can't stop us. Also some incidental cleanup on Zl1.

### CODE REVIEW

File: src/overlays/actors/ovl_Bg_Hidan_Fwbig/z_bg_hidan_fwbig.c [hunk 1]
```diff
@@ -21,29 +33,226 @@ const ActorInit Bg_Hidan_Fwbig_InitVars = {
...
+    this->actor.colChkInfo.mass = 0xFF;
+    this->direction = (u16)(thisx->params >> 8);
+    thisx->params &= 0xFF;
+    if (this->direction != 0) {
+        this->actor.initPosRot.pos.x = 1560.0f;
+        this->actor.initPosRot.pos.z = 0.0f;
+        if (player->actor.posRot.pos.z > 300.0f) {
+            this->direction = -1;
+            this->actor.shape.rot.y = -0x4E38;
+            this->actor.initPosRot.rot.y = this->actor.shape.rot.y;
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Hidan_Fwbig/z_bg_hidan_fwbig.c:64]:
```suggestion
            this->actor.initPosRot.rot.y = this->actor.shape.rot.y = -0x4E38;
```

File: src/overlays/actors/ovl_Bg_Hidan_Fwbig/z_bg_hidan_fwbig.c [hunk 2]
```diff
@@ -21,29 +33,226 @@ const ActorInit Bg_Hidan_Fwbig_InitVars = {
...
+        } else if (player->actor.posRot.pos.z < -300.0f) {
+            this->direction = 1;
+            this->actor.shape.rot.y = -0x31C8;
+            this->actor.initPosRot.rot.y = this->actor.shape.rot.y;
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Hidan_Fwbig/z_bg_hidan_fwbig.c:68]:
```suggestion
            this->actor.initPosRot.rot.y = this->actor.shape.rot.y = -0x31C8;
```

**Review by Roman971 — APPROVED** [review]

File: src/overlays/actors/ovl_Bg_Hidan_Fwbig/z_bg_hidan_fwbig.c [hunk 3]
```diff
@@ -21,29 +39,224 @@ const ActorInit Bg_Hidan_Fwbig_InitVars = {
...
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void BgHidanFwbig_UpdatePosition(BgHidanFwbig* this) {
+    s16 startAngle;
+
+    startAngle = this->actor.shape.rot.y + this->direction * -0x4000;
+
+    this->actor.posRot.pos.x = (Math_Sins(startAngle) * 885.4f) + this->actor.initPosRot.pos.x;
```
Comment (fig02) [inline src/overlays/actors/ovl_Bg_Hidan_Fwbig/z_bg_hidan_fwbig.c:102]: personally id just put this assignment with the declaration and remove the extra line break

File: src/overlays/actors/ovl_Bg_Hidan_Fwbig/z_bg_hidan_fwbig.c [hunk 4]
```diff
@@ -21,29 +39,224 @@ const ActorInit Bg_Hidan_Fwbig_InitVars = {
...
+            }
+            BgHidanFwbig_UpdatePosition(this);
+            this->actionFunc = BgHidanFwbig_Rise;
+        }
+    }
+}
+
+void BgHidanFwbig_WaitForTimer(BgHidanFwbig* this, GlobalContext* globalCtx) {
+    DECR(this->timer);
+    if (this->timer == 0) {
```
Comment (fig02) [inline src/overlays/actors/ovl_Bg_Hidan_Fwbig/z_bg_hidan_fwbig.c:154]: we're going to stay away from standalone DECRs for the time being

**Review by fig02 — APPROVED** [review]

**Pull request merged**